### ISSUE TITLE
Problems with the asp core at low load

### ISSUE THREAD

**Issue by EngRajabi** [dotnet/aspnetcore#27930]
### Describe the bug
I do not know in what category this issue is. But I recorded bugs.
When the api is written, it should call a server. If the server gets into trouble and runs out. Or slow down. Our service is slow and down.
[link repository code](https://github.com/EngRajabi/sample-httpClient)
Please consider the following tank.
The code that must happen for all requests is 500

requests: :10000
duration: 10s
status codes: {"0":960,"500":70,"503":8970}

### Exceptions (if any)

Loaded result

{"latencies":{"total":30145226477000,"mean":3014522647,"50th":1002607,"90th":15841351701,"95th":30001409856,"99th":30004296027,"max":30025735100,"min":996200},"bytes_in":{"total":2924220,"mean":292.422},"bytes_out":{"total":0,"mean":0},"earliest":"2020-11-17T20:06:34.0727973+03:30","latest":"2020-11-17T20:06:44.0717998+03:30","end":"2020-11-17T20:07:12.2181555+03:30","duration":9999002500,"wait":28146355700,"requests":10000,"rate":1000.0997599510551,"throughput":0,"success":0,"status_codes":{"0":960,"500":70,"503":8970},"errors":["503 Service Unavailable","500 Internal Server Error","Get \"http://localhost:8089/test\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)"]}

### Further technical details
- ASP.NET Core version: 5.0
- Include the output of `dotnet --info`: 5.0 latest
- The IDE (VS / VS Code/ VS4Mac) you're running on, and it's version: Visual Studio latest

**Comment by javiercn** [comment]
@EngRajabi thanks for contacting us.

Parking this in servers for now, since I'm not sure the right place for this, but @Tratcher or @davidfowl will have ideas here.

**Comment by BrennanConroy** [comment]
Are you running in IIS (IIS Express) or Kestrel?
Can you share any error logs?

What is the httpclient pointing at in your controller?
What would it require for us to run your sample?

**Comment by EngRajabi** [comment]
dotnet info
```

**Comment by EngRajabi** [comment]
dot net info
```
.NET SDK (reflecting any global.json):
 Version:   5.0.100
 Commit:    5044b93829

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19041
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.100\

Host (useful for support):
  Version: 5.0.0
  Commit:  cf258a14b7

.NET SDKs installed:
  2.1.809 [C:\Program Files\dotnet\sdk]
  2.2.300 [C:\Program Files\dotnet\sdk]
  3.0.103 [C:\Program Files\dotnet\sdk]
  3.1.201 [C:\Program Files\dotnet\sdk]
  5.0.100 [C:\Program Files\dotnet\sdk]
```

host: iis 10
hostingModel: inprocess
modules: AspNetCoreModuleV2

this is a report from my test:
requests: :10000
duration: 10s
status codes: {"0":960,"500":70,"503":8970}

To do my test, you need to remove the project source from the GitHub I posted in the first post. Build the project as a production and portable type.
Host in iis. Just call the method in the controller. This action wants to connect to a service that has been downgraded.

**Comment by Tratcher** [comment]
So this [line](https://github.com/EngRajabi/sample-httpClient/blob/64ae1bcb341ff11f2e53a9dfcf539face5833f8b/Core5/Controllers/TestController.cs#L32) is throwing an exception because `http://localhost:8080` is down? Which exception is being thrown? How long do these calls take on average before they fail?

The 503s would be happening because the Http.Sys queue defaults to 1000, fills up, and rejects additional requests. The 960+70 = 1030 requests make sense for about how many requests it's able to queue and start processing during the 10s test window. IIS also slowly ramps up the number of threads it uses to process requests in the queue, so if you raised the queue limit or ran a longer test you may see it start to catch up.

**Comment by EngRajabi** [comment]
Yes, this error creates an exception because another service is down or disconnected.
Connection refuse error occurs. It takes about 1 to 2 seconds for this exception to happen.
I even increased the number of queues from 1000 to 10,000 on iis. I also increased the time from 10 seconds to 15.
But it did not matter.

new load result:
```
{"latencies":{"total":300268047519600,"mean":20017869834,"50th":30000085389,"90th":30001092022,"95th":30003048816,"99th":30007021336,"max":30016080500,"min":1000100},"bytes_in":{"total":1616960,"mean":107.79733333333333},"bytes_out":{"total":0,"mean":0},"earliest":"2020-11-20T10:51:16.5506596+03:30","latest":"2020-11-20T10:51:31.5497486+03:30","end":"2020-11-20T10:51:58.8405863+03:30","duration":14999089000,"wait":27290837700,"requests":15000,"rate":1000.0607370220952,"throughput":0,"success":0,"status_codes":{"0":9970,"500":70,"503":4960},"errors":["500 Internal Server Error","503 Service Unavailable","Get \"http://localhost:8089/test\": context deadline exceeded (Client.Timeout exceeded while awaiting headers)"]}
```
